core/local/local.go

```diff
@@ -224,6 +224,30 @@ func (e *ExecutionScheduler) initVUsConcurrently(
 func (e *ExecutionScheduler) Init(ctx context.Context, samplesOut chan<- stats.SampleContainer) error {
 	logger := e.logger.WithField("phase", "local-execution-scheduler-init")
 
+	// Initialize each executor and do some basic validation.
+	e.state.SetExecutionStatus(lib.ExecutionStatusInitExecutors)
+	logger.Debugf("Start initializing executors...")
+	errMsg := "error while initializing executor %s: %s"
+	exports := e.runner.GetExports()
+	for _, exec := range e.executors {
+		executorConfig := exec.GetConfig()
+		execFn := executorConfig.GetExec().ValueOrZero()
+		execName := executorConfig.GetName()
+
+		if execFn == "" {
+			execFn = "default"
+		}
+		if _, ok := exports[execFn]; !ok {
+			return fmt.Errorf(errMsg, execName,
+				fmt.Sprintf("function '%s' not found in exports", execFn))
+		}
+		if err := exec.Init(ctx); err != nil {
+			return fmt.Errorf(errMsg, executorConfig.GetName(), err)
+		}
+		logger.Debugf("Initialized executor %s", executorConfig.GetName())
+	}
+
+	logger.Debugf("Finished initializing executors, start initializing VUs...")
```

Comment:
I moved the executor init before the VU init to fail quicker in case there's nothing to run.

Reply:
hmm I don't think executor initialization would require initialized VUs, so I don't see a reason not to do it like this :+1:

Reply:
hmm₂ - wondering if we shouldn't do the validation in `deriveAndValidateConfig` - we have the exported function by that time, we just have to pass them in there?

Reply:
Good idea about doing the validation earlier! This should be resolved in a1859244.